firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -0,0 +1,89 @@
+
+
+#include "LPC17xx.h"
+
+class SSP
+{
+private: 
+	/*SSP register lookup table*/
+	LPC_SSP_TypeDef* SSP[3] = {LPC_SSP0, LPC_SSP1, LPC_SSP2};

```

Comment:
See PinConfigure. You want to make this static and initialize in a cpp file for the purpose of dependency injection.